scripts/build_connectome/func_connectivity.py

```diff
@@ -21,123 +22,196 @@
-                                  f'dir-{direction}.csv'))
-            np.savetxt(corr, correlation_matrix, delimiter=',')
-            fig = plt.figure(figsize=(10, 10))
-            plotting.plot_matrix(correlation_matrix, labels=atlas.labels,
-                                 figure=fig, vmax=1, vmin=-1,
-                                 title='Covariance')
-            corr_fig = os.path.join(tmp_dir, (f'{atlas.name}_corr_{sub}_{ses}_'
-                                              f'dir-{direction}.png'))
-            fig.savefig(corr_fig, bbox_inches='tight')
-            
-            # define a sparse inverse covariance estimator
-            glc.fit(time_series)
-    
-            # save correlation and partial correlation matrices as csv
-            part_corr = os.path.join(tmp_dir, (f'{atlas.name}_part_corr_{sub}_'
-                                               f'{ses}_dir-{direction}.csv'))
-            np.savetxt(part_corr, -glc.precision_, delimiter=',')
-    
-            # plot heatmaps and save figs                     
-            fig = plt.figure(figsize = (10, 10))
-            plotting.plot_matrix(-glc.precision_, labels=atlas.labels,
-                                 figure=fig, vmax=1, vmin=-1,
-                                 title='Sparse inverse covariance')
-            part_corr_fig = os.path.join(tmp_dir, 
-                                         (f'{atlas.name}_part_corr_{sub}_{ses}_'
-                                          f'dir-{direction}.png'))
-            fig.savefig(part_corr_fig, bbox_inches='tight')
-
-    # Use sparse group inverse across 4 runs for beter estimates
-    # gsc.fit(all_time_series)
-    
-    glc.fit(np.concatenate(all_time_series))
-    # save correlation and partial correlation matrices as csv
-    part_corr = os.path.join(tmp_dir, f'{atlas.name}_part_corr_{sub}_all_compcorr.csv')
-    np.savetxt(part_corr, -glc.precision_, delimiter=',')
-    corr = os.path.join(tmp_dir, f'{atlas.name}_corr_{sub}_all.csv')
-    np.savetxt(corr, glc.covariance_, delimiter=',')
-    
-    # plot heatmaps and save figs                     
-    fig = plt.figure(figsize = (10, 10))
-    plotting.plot_matrix(-glc.precision_, labels=atlas.labels,
-                         figure=fig, vmax=1, vmin=-1,
-                         title='Sparse inverse covariance')
-    part_corr_fig = os.path.join(
-        tmp_dir, f'{atlas.name}_part_corr_{sub}_all.png')
-    fig.savefig(part_corr_fig, bbox_inches='tight')
-    #
-    fig = plt.figure(figsize = (10, 10))
-    plotting.plot_matrix(glc.covariance_, labels=atlas.labels,
-                         figure=fig, vmax=1, vmin=-1,
-                         title='Covariance with sparse inverse')
-    corr_fig = os.path.join(
-        tmp_dir, f'{atlas.name}_corr_{sub}_all.png')
-    fig.savefig(corr_fig, bbox_inches='tight')
-    plt.close('all')
+
+for fmri in fmris:
+    if fmri == "movie":
```

Comment:
It would be great if the code could be reorganized a little bit to avoid the unreadable `if` loop.
LGTM otherwise.